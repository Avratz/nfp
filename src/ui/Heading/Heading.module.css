.heading {
  font-family: var(--font-dmSans);
  font-weight: var(--font-bold);
}

.2xs {
  font-size: var(--text-2xs);
  line-height: var(--line-2xs);
}

.xs {
  font-size: var(--text-xs);
  line-height: var(--line-xs);
}

.sm {
  font-size: var(--text-sm);
  line-height: var(--line-sm);
}

.base {
  font-size: var(--text-base);
  line-height: var(--line-base);
}

.lg {
  font-size: var(--text-lg);
  line-height: var(--line-lg);
}

.xl {
  font-size: var(--text-xl);
  line-height: var(--line-xl);
}

.2xl {
  font-size: var(--text-2xl);
  line-height: var(--line-2xl);
}

.3xl {
  font-size: var(--text-3xl);
  line-height: var(--line-3xl);
}

.4xl {
  font-size: var(--text-4xl);
  line-height: var(--line-4xl);
}

.5xl {
  font-size: var(--text-5xl);
  line-height: var(--line-5xl);
}

.6xl {
  font-size: var(--text-6xl);
  line-height: var(--line-6xl);
}

.inlineBlock {
  display: inline-block;
}
