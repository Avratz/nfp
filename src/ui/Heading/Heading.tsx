import { createElement, forwardRef, ReactNode } from 'react'
import { cva, VariantProps } from 'class-variance-authority'

import styles from './Heading.module.css'

const headingStyles = cva(styles.heading, {
  variants: {
    size: {
      '2xs': styles['2xs'],
      xs: styles.xs,
      sm: styles.sm,
      base: styles.base,
      lg: styles.lg,
      xl: styles.xl,
      '2xl': styles['2xl'],
      '3xl': styles['3xl'],
      '4xl': styles['4xl'],
      '5xl': styles['5xl'],
      '6xl': styles['6xl'],
    },
    inlineBlock: {
      true: styles.inlineBlock,
    },
  },
  defaultVariants: {
    size: 'base',
    inlineBlock: false,
  },
})

interface HeadingProps extends VariantProps<typeof headingStyles> {
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | undefined
  children: ReactNode
  className?: string
  inlineBlock?: boolean
}

export const Heading = forwardRef<HTMLHeadingElement, HeadingProps>(
  function forwardHeading(
    { as = 'h1', size = 'base', children, className, inlineBlock },
    ref,
  ) {
    return createElement(
      as,
      {
        ref,
        className: headingStyles({ size, inlineBlock, className }),
      },
      children,
    )
  },
)
