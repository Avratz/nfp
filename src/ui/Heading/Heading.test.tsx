import { render } from '@testing-library/react'
import { Heading } from './Heading'

describe('Heading', () => {
  it('renders default', () => {
    const { container } = render(<Heading>Heading</Heading>)
    expect(container).toMatchSnapshot()
  })

  it('renders as h2 with size xl', () => {
    const { container } = render(
      <Heading as="h2" size="xl">
        Heading
      </Heading>,
    )
    expect(container).toMatchSnapshot()
  })
})
