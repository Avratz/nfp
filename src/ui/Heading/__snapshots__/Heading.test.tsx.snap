// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders as h2 with size xl 1`] = `
<div>
  <h2
    class="heading xl"
  >
    Heading
  </h2>
</div>
`;

exports[`Heading renders default 1`] = `
<div>
  <h1
    class="heading base"
  >
    Heading
  </h1>
</div>
`;
